go-controller/pkg/ovn/policy.go

```diff
@@ -892,15 +893,31 @@ func (oc *Controller) handlePeerNamespaceSelector(
 				np.Lock()
 				defer np.Unlock()
 				if !np.deleted {
-					gress.addNamespaceAddressSet(namespace.Name, np.portGroupName)
+					//FIXME :  Dual stack . We need to first fix ACL generation in case of dual stack
+					// Then we will remove the if logic and add both address sets.
+					// Will be fixed in next set of PRs
+
+					if config.IPv6Mode {
+						gress.addNamespaceAddressSet(getIPv6AddressSetName(namespace.Name), np.portGroupName)
```

Comment:
So there's `config.IPv4Mode` too. What you want is to construct an ACL where in ipv4-only mode you check "`ip4.src == $IP4_ADDRESS_SET`" and in ipv6-only mode you check "`ip6.src == $IP6_ADDRESS_SET`" and in dual-stack mode (`config.IPv6Mode && config.IPv4Mode`) you check "`(ip4.src == $IP4_ADDRESS_SET || ip6.src == $IP6_ADDRESS_SET)`"